util/compiler.h

```diff
@@ -5,11 +5,13 @@
 /* Use to tag a variable that causes compiler warnings. Use as:
     int uninitialized_var(sz)
 
-   This is only enabled for old compilers. gcc 6.x and beyond have excellent
-   static flow analysis. If code solicits a warning from 6.x it is almost
-   certainly too complex for a human to understand.
+   This is only enabled for old compilers and non-x86 architectures. gcc 6.x
+   and beyond have excellent static flow analysis for x86 platforms. If code
+   solicits a warning from 6.x it is almost certainly too complex for a human to
+   understand.
 */
-#if __GNUC__ >= 6 || defined(__clang__)
+#if (__GNUC__ >= 6 && !defined(__aarch64__)) || \
+    defined(__clang__)
 #define uninitialized_var(x) x
```

Comment:
That should be documented either in the code or in the commit message because otherwise anyone who sees " && !defined(__aarch64__)" will wonder why it's there ...